APCAppCore/APCAppCore/Library/Categories/HKWorkout+APCHelper.m

```diff
@@ -0,0 +1,111 @@
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+#import "HKWorkout+APCHelper.h"
+
+@implementation HKWorkout (APCHelper)
+
++ (NSString*)apc_workoutActivityTypeStringRepresentation:(int)num
+{
+    static NSDictionary* activityEvents;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^
+    {
+        activityEvents =
+        @{
+        @(1)    : @"HKWorkoutActivityTypeAmericanFootball",
+        @(2)    : @"HKWorkoutActivityTypeArchery",
+        @(3)    : @"HKWorkoutActivityTypeAustralianFootball",
+        @(4)    : @"HKWorkoutActivityTypeBadminton",
+        @(5)    : @"HKWorkoutActivityTypeBaseball",
+        @(6)    : @"HKWorkoutActivityTypeBasketball",
+        @(7)    : @"HKWorkoutActivityTypeBowling",
+        @(8)    : @"HKWorkoutActivityTypeBoxing",
+        @(9)    : @"HKWorkoutActivityTypeClimbing",
+        @(10)   : @"HKWorkoutActivityTypeCricket",
+        @(11)   : @"HKWorkoutActivityTypeCrossTraining", // Any mix of cardio and/or strength and/or flexibility
+        @(12)   : @"HKWorkoutActivityTypeCurling",
+        @(13)   : @"HKWorkoutActivityTypeCycling",
+        @(14)   : @"HKWorkoutActivityTypeDance",
+        @(15)   : @"HKWorkoutActivityTypeDanceInspiredTraining", // Pilates, Barre, Feldenkrais, etc.
+        @(16)   : @"HKWorkoutActivityTypeElliptical",
+        @(17)   : @"HKWorkoutActivityTypeEquestrianSports", // Polo, Horse Racing, Horse Riding, etc.
```

Comment:
Hm, ok. Would be better to use a switch without a default: case for generating these mappings, so that if new workout activity type is added, you catch it. Could also use a macro so that the enum name is guaranteed to exactly match the string.